src/models/http/baseHttpServer.js

```diff
@@ -32,13 +32,28 @@ function setup (protocolName, createBaseServer) {
                     body: stubResponse.body || defaultResponse.body || '',
                     _mode: stubResponse._mode || defaultResponse._mode || 'text'
                 },
+                responseHeaders = headersHelper.getJar(response.headers),
                 encoding = response._mode === 'binary' ? 'base64' : 'utf8';
 
             if (typeof response.body === 'object') {
                 // Support JSON response bodies
                 response.body = JSON.stringify(response.body, null, 4);
             }
 
+            if (options.allowCORS) {
+                var requestHeaders = headersHelper.getJar(request.headers),
+                    isCrossOriginPreflight = request.method === 'OPTIONS' &&
+                        requestHeaders.get('Access-Control-Request-Headers') &&
+                        requestHeaders.get('Access-Control-Request-Method') &&
+                        requestHeaders.get('Origin');
+
+                if (isCrossOriginPreflight) {
+                    responseHeaders.set('Access-Control-Allow-Headers', requestHeaders.get('Access-Control-Request-Headers'));
```

Comment:
Here I'm not sure whether I should check whether the headers already have a value? Maybe I should only set a header's value if has no value?